segmentation_models_pytorch/decoders/unet/model.py

```diff
@@ -65,20 +65,30 @@ def __init__(
         classes: int = 1,
         activation: Optional[Union[str, callable]] = None,
         aux_params: Optional[dict] = None,
+        encoder_kwargs: Optional[dict] = None,
     ):
         super().__init__()
 
+        # if sam encoder, make sure to make num_hidden_skips is set
+        if encoder_name.startswith("sam-"):
+            encoder_kwargs = encoder_kwargs if encoder_kwargs is not None else {}
+            encoder_kwargs.update({"num_hidden_skips": len(decoder_channels)})
+            n_decoder_blocks = len(decoder_channels)
+        else:
+            n_decoder_blocks = encoder_depth
+
```

Comment:
is it possible to make SAM encoder responsible for this logic?
I am trying to keep API simple and unified, if there is an option not to pass kwargs it is better to do it this way

Reply:
the sole purpose of this is to provide a correct dummy output for Unet decoder.